src/Core/Content/Flow/Rule/OrderTransactionStatusRule.php

```diff
@@ -52,27 +53,39 @@ public function match(RuleScope $scope): bool
             return false;
         }
 
-        if (!$transactions = $scope->getOrder()->getTransactions()) {
-            return false;
-        }
+        if (!Feature::isActive('v6.8.0.0')) {
+            if (!$transactions = $scope->getOrder()->getTransactions()) {
+                return false;
+            }
 
-        /** @var OrderTransactionEntity $last */
-        $last = $transactions->last();
-        $paymentMethodId = $last->getStateId();
+            /** @var OrderTransactionEntity $last */
+            $last = $transactions->last();
+            $paymentMethodId = $last->getStateId();
 
-        foreach ($transactions->getElements() as $transaction) {
-            $technicalName = $transaction->getStateMachineState()?->getTechnicalName();
-            if ($technicalName !== null
-                && $technicalName !== OrderTransactionStates::STATE_FAILED
-                && $technicalName !== OrderTransactionStates::STATE_CANCELLED
-            ) {
-                $paymentMethodId = $transaction->getStateId();
+            foreach ($transactions->getElements() as $transaction) {
+                $technicalName = $transaction->getStateMachineState()?->getTechnicalName();
+                if ($technicalName !== null
+                    && $technicalName !== OrderTransactionStates::STATE_FAILED
+                    && $technicalName !== OrderTransactionStates::STATE_CANCELLED
+                ) {
+                    $paymentMethodId = $transaction->getStateId();
 
-                break;
+                    break;
+                }
             }
+
+            return RuleComparison::stringArray($paymentMethodId, $this->stateIds, $this->operator);
+        }
+
+        if (!$scope->getOrder()->getPrimaryOrderTransaction()) {
```

Comment:
I'm not sure if this is correct. Before, the last non-cancelled and failed transaction was used, whereas now the primary is used, which could be a failed or cancelled one :thinking: